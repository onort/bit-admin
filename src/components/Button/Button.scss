@import "../../styles/main.scss";

.button {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: px-to-rem(8px) px-to-rem(12px);
  margin-right: $spacer-md;
  font-size: $font-size-md;
  color: $text-color;
  border: 0;
  border-radius: px-to-rem(3px);
  background-color: $light-bg;
  cursor: pointer;
  transition-duration: 0.3s;

  &:hover {
    background-color: darken($light-bg, 5);
  }
}

.success {
  background-color: $success;
  color: $white;
  &:hover {
    background-color: lighten($success, 5);
  }
}

.danger {
  background-color: $danger;
  color: $white;
  &:hover {
    background-color: lighten($danger, 5);
  }
}

.text {
  display: block;
}

.icon {
  display: flex;
  font-size: $font-size-lg;
  margin-right: $spacer-xsm;
}
