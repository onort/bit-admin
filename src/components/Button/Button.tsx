import React from "react"
import cx from "classnames"

import styles from "./Button.scss"

type ButtonHTMLType = "submit" | "button" | "reset"

interface Props {
  className?: string
  icon?: React.ReactNode
  onClick?: (e: React.MouseEvent<HTMLElement>) => void
  text?: string
  type?: ButtonHTMLType
}

const Button: React.SFC<Props> = props => {
  const className = cx(styles.button, props.className)
  return (
    <button type={props.type} className={className} onClick={props.onClick}>
      {props.icon && <span className={styles.icon}>{props.icon}</span>}
      <span className={styles.text}>{props.text}</span>
    </button>
  )
}
export default Button
