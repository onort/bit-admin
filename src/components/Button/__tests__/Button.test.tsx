import React from "react"
import { shallow } from "enzyme"
import { MdAdd as AddIcon } from "react-icons/md"

import Button from "../"

const testText = "Test"
const testClassName = "testClassName"

describe("<Button />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<Button text={testText} />)
    expect(wrapper).toMatchSnapshot()
  })

  it("should have text rendered", () => {
    const wrapper = shallow(<Button text={testText} />)
    expect(wrapper.find("button").text()).toBe(testText)
  })

  it("should render an icon component if passed through prop", () => {
    const wrapper = shallow(<Button text={testText} icon={<AddIcon />} />)
    expect(wrapper.find(AddIcon).exists()).toBe(true)
  })

  it("should be able to have a custom className", () => {
    const wrapper = shallow(
      <Button text={testText} className={testClassName} />
    )
    expect(wrapper.find("button").hasClass(testClassName)).toBe(true)
  })

  it("should fire onClick when button is clicked", () => {
    const handleClick = jest.fn()
    const wrapper = shallow(<Button text={testText} onClick={handleClick} />)
    wrapper.find("button").simulate("click")
    expect(handleClick).toBeCalledTimes(1)
  })
})
