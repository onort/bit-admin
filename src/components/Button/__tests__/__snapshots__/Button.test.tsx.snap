// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot 1`] = `
<button
  className="button"
>
  <span
    className="text"
  >
    Test
  </span>
</button>
`;
